src/library/scala/collection/immutable/ChampHashMap.scala

```diff
@@ -570,43 +860,69 @@ private final class BitmapIndexedMapNode[K, +V](val dataMap: Int, val nodeMap: I
 
 }
 
-private final class HashCollisionMapNode[K, +V](val originalHash: Int, val hash: Int, val content: Vector[(K, V)]) extends MapNode[K, V] {
+private final class HashCollisionMapNode[K, +V ](
+  val originalHash: Int,
+  val hash: Int,
+  var content: Vector[(K, V @uV)]
```

Comment:
wouldn't it be better to maintain the content as Array/Vector[AnyRef]
this reduces the allocation of tuples, and is more similar to BitmapIndexedMapNode

I am not convinced that Vector is providing and structural sharing here (maybe where we have a collision that has >32 collisions already and we add another one)
If structural sharing is not useful then shouln't this be an Array to reduce the footprint. Maybe its a sperate PR though as the change is orthogonal to this one

Reply:
I think this is like you say really important, and worthy of an other discussion

Reply:
@mkeskells see https://github.com/scala/bug/issues/11164